WooCommerce/Classes/ViewRelated/Orders/ScanToPay/ScanToPayViewModel.swift

```diff
@@ -0,0 +1,28 @@
+import Foundation
+import CoreImage.CIFilterBuiltins
+import UIKit
+
+struct ScanToPayViewModel {
+    private let paymentURL: URL?
+
+    init(paymentURL: URL?) {
+        self.paymentURL = paymentURL
+    }
+
+    func generateQRCodeImage() -> UIImage? {
+        guard let paymentURLString = paymentURL?.absoluteString else {
+            return nil
+        }
+
+        let context = CIContext()
+        let filter = CIFilter.qrCodeGenerator()
+        filter.message = Data(paymentURLString.utf8)
+
+        guard let outputImage = filter.outputImage,
+              let cgImage = context.createCGImage(outputImage, from: outputImage.extent) else {
+            return nil
+        }
+
+        return UIImage(cgImage: cgImage)
+    }
```

Comment:
Should we abstract and extract this method somewhere else? Let's say an extension of UIImage, or as an Utility method in WooFoundation, then change the signature to something like the following `generateQRCodeFromPaymentDetails(paymentURLString: URL, message: String) -> UIImage?`

This way the view model doesn't have to know about the QR generation implementation details, we just pass the URL, the message, and let the system return it.

Reply:
Good point, I also considered that and preferred leaving it in the view model. The rationale is that at the moment we only generate a QR code here, and the function is simple enough to keep it in the view model. Furthermore, generating an image that we're going to display in the view from model data matches the responsibility of a view model.

If more logic is added to the QR code generation, e.g., adding the woo logo to it, I will extract it to its own class/extension.